packages/odyssey-react-mui/src/labs/DataFilters.tsx

```diff
@@ -0,0 +1,540 @@
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import {
+  MutableRefObject,
+  ReactNode,
+  memo,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { Box } from "../Box";
+import { TagList } from "../TagList";
+import { Tag } from "../Tag";
+import { SearchField } from "../SearchField";
+import { Button } from "../Button";
+import { IconButton, Menu, Popover } from "@mui/material";
+import {
+  CheckIcon,
+  ChevronRightIcon,
+  CloseCircleFilledIcon,
+  FilterIcon,
+} from "../icons.generated";
+import { MenuItem } from "../MenuItem";
+import { Paragraph, Subordinate } from "../Typography";
+import { TextField } from "../TextField";
+import { CheckboxGroup } from "../CheckboxGroup";
+import { Checkbox } from "../Checkbox";
+import { RadioGroup } from "../RadioGroup";
+import { Radio } from "../Radio";
+
+export const dataFilterVariantValues = [
+  "autocomplete",
+  "checkbox",
+  "date",
+  "date-range",
+  "multi-select",
+  "range",
+  "range-slider",
+  "select",
+  "text",
+] as const;
+
+export type DataFilter = {
+  id: string;
+  label: string;
+  variant?: (typeof dataFilterVariantValues)[number];
+  value?: string | string[] | undefined;
+  options?: Array<{ label: string; value: string }>;
+};
+
+export type DataFiltersProps = {
+  onChangeSearch?: (value: string) => void;
+  onChangeFilters?: (filters: Array<DataFilter>) => void;
+  searchOnSubmit?: boolean;
+  searchDelay?: number;
+  initialSearchTerm?: string;
+  additionalActions?: ReactNode;
+  filters?: Array<DataFilter>;
+};
+
+const DataFilters = ({
+  onChangeSearch,
+  onChangeFilters,
+  searchOnSubmit = false,
+  searchDelay = 200,
+  initialSearchTerm,
+  additionalActions,
+  filters: filtersProp = [],
+}: DataFiltersProps) => {
+  const [filters, setFilters] = useState<DataFilter[]>(filtersProp);
+  const [inputValues, setInputValues] = useState<{
+    [key: string]: string | string[] | undefined;
+  }>(
+    filtersProp.reduce(
+      (
+        accumulator: { [key: string]: string | string[] | undefined },
+        filter: DataFilter
+      ) => {
+        accumulator[filter.id] = filter.value;
+        return accumulator;
+      },
+      {}
+    )
+  );
+  const [search, setSearch] = useState<string>(initialSearchTerm ?? "");
+  const activeFilters = filters.filter((filter) => filter.value);
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const [menuAnchorEl, setMenuAnchorEl] = useState<HTMLElement | undefined>(
+    undefined
+  );
```

Comment:
```suggestion
  const [menuAnchorElement, setMenuAnchorElement] = useState<HTMLElement | undefined>(
    undefined
  );
```

_Which_ `menuAnchorElement` is this? Why can it sometimes be `undefined`? I can't tell from reading this part of the code. Is it the selected menu anchor element? Is it the only one? Based on the way it's written, I would expect there to only be one, but then why is there a state?